src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -229,7 +204,7 @@ private void Collect()
                 processor.OnEnd(metricItem);
             }
 
-            foreach (var processor in this.ExportProcessors)
+            foreach (var processor in processors)
```

Comment:
what does `this.MetricProcessors` do? is exportProcessor is just a special type of MetricProcessor? Does the meterproviderSdk need to care about both MetricProcessors, and ExportProcessors separately?

Reply:
Metric Processor allows for filtering and alteration of Metrics before it gets to the "endpoint" Exporter. MetricProcessor takes in MetricItems and output MetricItems. Exporter takes in MetricItems, but is the endpoint so will not output anything.

In Abstract, they are all MetricProcessors, but Exporters are specialized version of MetricProcessors, and needs special treatment in the staging and ordering of when it runs.

Reply:
See Spec. It has both MetricProcessor and MetricExporter.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricprocessor